src/mirrored/linux.rs

```diff
@@ -71,15 +71,16 @@ pub fn allocate_mirrored(size: usize) -> Result<*mut u8, AllocError> {
         assert!(half_size % allocation_granularity() == 0);
 
         // create temporary file
-        let mut fname = *b"/tmp/slice_deque_fileXXXXXX\0";
-        let mut fd: c_long =
-            memfd_create(fname.as_mut_ptr() as *mut c_char, 0);
+        let fname = b"/tmp/slice_deque_fileXXXXXX\0";
```

Comment:
If I understood the problem correctly (which I'm not sure), _I think_ the current code might be correct, since it is just equivalent to:

```suggestion
        let mut fname: [u8; 29] = b"/tmp/slice_deque_fileXXXXXX\0";
```

Which arguably is much clearer than how the code is currently written.

Reply:
For example, this code in the [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=26ef206296a5e33d3c85c0bd8c0d7677) compiles:

```rust
   let mut fname = *b"/tmp/slice_deque_fileXXXXXX\0";
   let mut fname2: [u8; 28] = fname;
```

So the type of `fname` in the current version of the code is indeed `[u8; 28]`, which is a local copy on the stack of the string literal.